Client/Frontend/Browser/URIFixup.swift

```diff
@@ -34,4 +39,29 @@ class URIFixup {
 
         return nil
     }
-}
\ No newline at end of file
+
+    static func strToAscii(string : String) -> String {
+        var host : String?
+
+        if string.hasPrefix("http://") {
+            host = string.substringFromIndex(string.startIndex.advancedBy(7))
+        } else if string.hasPrefix("https://") {
+            host = string.substringFromIndex(string.startIndex.advancedBy(8))
+        } else {
+            host = string
+        }
+
+        let t = host!.rangeOfString("/")
+        if (t?.startIndex) != nil {
+            host = host?.substringToIndex((t?.startIndex)!)
+        }
+
+        let s = host!.rangeOfString(":")
+        if (s?.startIndex) != nil {
+            host = host?.substringToIndex((t?.startIndex)!)
+        }
+
+        let urlStr = host?.utf8HostToAscii()
+        return urlStr!
```

Comment:
On line 36 I must use `punycodedURL("http://(escaped!)")`.

I have changed this code according to your suggestion.

Do you have any other suggestion?